### PR TITLE
Modified basic-syscall and syscall-wrappers infrastructure

### PR DESCRIPTION
**Pull request by SorinAlexB**
Added checkers and modified files

# Prerequisite Checklist

- [ ] Read the [contribution guidelines](https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [ ] Tested your changes against relevant architectures and platforms;
- [ ] Updated relevant documentation (if needed).

### CODE REVIEW

File: chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/tests/checker.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ref="ref/out.ref"
+
+head -n 3 "$out" > /tmp/out.txt
+cat /tmp/out.txt
+
+cmp -s "$ref" /tmp/out.txt
+if [ $? -eq 0 ]; then
+    echo "main ...................... passed ... 10"
+else
+    echo "main ...................... failed ... 0"
```
Comment (teodutu) [inline chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/tests/checker.sh:19]: This only tests `read()` and `write()`. To test `itoa()` and `reverse_string()`, change the former's name to `os_itoa()` or something and compare their outputs to `itoa()` and `strrev()`. You can test `getpid()` by changing its name in `syscall.asm` to `os_getpid()` and comparing its output to the real `getpid()`.

File: chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/src/main.c
Comment (teodutu) [inline chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/src/main.c:1]: Change the `main()` function to test `itoa()` and `reverse_string()` separately before testing `getpid()` using `itoa()`.

**Review by teodutu — CHANGES_REQUESTED** [review]
Having the tests only in the `solution/` folder doesn't seem to be the way to go. Add a command to the makefiles so that `make skels` also copies the tests to the `support/` folder.

In addition, fix the liter failures.

File: chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/tests/checker.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+mkdir -p out
+touch out/out1.out
+touch out/out2.out
+touch out/out3.out
+
+out="out/out.out"
+ref="ref/out.ref"
+
+head -n 3 "$out" > /tmp/out.txt
```
[the remark this answers is not in the document]
Comment (SorinAlexB) [reply chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/tests/checker.sh:12]: Students should implement the getpid syscall wrapper. I assumed it should not be tested, so the checker evaluates just the first three lines of output(read and write syscalls).

[the remark this answers is not in the document]
Comment (teodutu) [reply chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/tests/checker.sh:12]: It should be tested. Students need a way to be sure they're using the right syscall number.

**Comment by teodutu** [comment]
Fuck me. TIL that `strrev()` isn't supported by GCC :upside_down_face: [1]. And you can use `snprintf()` as an implementation for `itoa()` [2].

[1] https://stackoverflow.com/a/58898320
[2] https://stackoverflow.com/a/12970536

File: chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/tests/checker.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+if cmp -s "$ref" /tmp/out.txt; then
+    echo "main ...................... passed ... 10"
+else
+    echo "main ...................... failed ... 0"
+fi
```
Comment (teodutu) [inline chapters/software-stack/system-calls/drills/tasks/syscall-wrapper/solution/tests/checker.sh:17]: You should make this test more granular: test each functionality individually, not all of them at once.

**Review by teodutu — CHANGES_REQUESTED** [review]
Just add the `make` commands to the checkers and update the paths to the binaries. Regarding testing `syscall-wrapper` granularly, let's leave it as it is for now and I'll convert this into an issue for a later time so we can merge this now.

**Review by teodutu — APPROVED** [review]

**Pull request merged**